lib/include/kickcat/CoE/OD.h

```diff
@@ -0,0 +1,191 @@
+        ARRAY_OF_SINT   = 0x0261,   // ARRAY [0..n] OF SINT
+        ARRAY_OF_DINT   = 0x0262,   // ARRAY [0..n] OF DINT
+        ARRAY_OF_UDINT  = 0x0263,   // ARRAY [0..n] OF UDINT
+
+        PDO_MAPPING     = 0x0021,
+        SDO_PARAMETER   = 0x0022,
+        IDENTITY        = 0x0023,
+
+        COMMAND_PAR            = 0x0025,
+        PDO_PARAMETER          = 0x0027,
+        ENUM                   = 0x0028,
+        SM_SYNCHRONISATION     = 0x0029,
+        RECORD                 = 0x002A,
+        BACKUP_PARAMETER       = 0x002B,
+        MODULAR_DEVICE_PROFILE = 0x002C,
+        ERROR_SETTING          = 0x0281,
+        DIAGNOSIS_HISTORY      = 0x0282,
+        EXTERNAL_SYNC_STATUS   = 0x0283,
+        EXTERNAL_SYNC_SETTINGS = 0x0284,
+        DEFTYPE_FSOEFRAME      = 0x0285,
+        DEFTYPE_FSOECOMMPAR    = 0x0286
+    };
+    char const* toString(enum DataType type);
+    constexpr bool isBasic(DataType type)
+    {
+        switch (type)
+        {
+            case DataType::INTEGER8:
+            case DataType::INTEGER16:
+            case DataType::INTEGER32:
+            case DataType::INTEGER64:
+            case DataType::UNSIGNED8:
+            case DataType::UNSIGNED16:
+            case DataType::UNSIGNED32:
+            case DataType::UNSIGNED64:
+            case DataType::REAL32:
+            case DataType::REAL64:
+            case DataType::BOOLEAN:
+            case DataType::BYTE:
+            {
+                return true;
+            }
+            default:
+            {
+                return false;
+            }
+        }
+    }
+
+    // ETG1000.5 Chapter 6.1.4.2.1 Formal model
+    // ETG1000.6 Chapter 5.6.3.6.2 Get Entry Description Response
+    namespace Access
+    {
+        constexpr uint16_t READ_PREOP   = (1 << 0);
+        constexpr uint16_t READ_SAFEOP  = (1 << 1);
+        constexpr uint16_t READ_OP      = (1 << 2);
+        constexpr uint16_t WRITE_PREOP  = (1 << 3);
+        constexpr uint16_t WRITE_SAFEOP = (1 << 4);
+        constexpr uint16_t WRITE_OP     = (1 << 5);
+        constexpr uint16_t RxPDO        = (1 << 6);
+        constexpr uint16_t TxPDO        = (1 << 7);
+        constexpr uint16_t BACKUP       = (1 << 8);
+        constexpr uint16_t SETTING      = (1 << 9);
+
+        // helpers
+        constexpr uint16_t READ     = (Access::READ_PREOP  | Access::READ_SAFEOP  | Access::READ_OP);
+        constexpr uint16_t WRITE    = (Access::WRITE_PREOP | Access::WRITE_SAFEOP | Access::WRITE_OP);
+        constexpr uint16_t MAPPABLE = (Access::RxPDO | Access::TxPDO);
+
+        std::string toString(uint16_t access);
+    };
+
+    struct Entry    // ETG1000.5 6.1.4.2.1 Formal model
+    {
+        Entry() = default;
+        ~Entry();
+
+        Entry(Entry const&) = delete;
+        Entry& operator=(Entry const& other) = delete;
+
+        Entry(Entry&& other);
+        Entry& operator=(Entry&& other);
+
+        uint8_t      subindex;
```

Comment:
In table 48 "Get Entry Description Response" ETG1000.6 it specifies also the value info and datatype before bit len.
And no description.

There seems to be changes of order/mismatch with this table.

Reply:
This entry is NOT the answer of the SDO, but an internal storage: it has to be serialized for such purpose.

Doing this way enable non POD management (so constructor/destructor, std::string or STL container, etc). It seems more important to have a good itnernal class than an already 'mappable' version for SDO Info service.